### PR TITLE
Add NavX to the DriveSubsystem

### PR DESCRIPTION
**Pull request by mandrews281**
Added the creation & calibration of the NavX to the RealDriveSubsystem.  Also added the periodic method so we can see if it's working

### CODE REVIEW

File: Template2018/src/main/java/frc/team281/robot/subsystems/RealDriveSubsystem.java [hunk 1]
```diff
@@ -145,22 +168,31 @@ public void tankDrive(double left, double right) {
 		}
 	}
 
+	@Override
+	public void periodic() {
+        // method called by scheduler automatically
+		if (this.navX != null) {
+            SmartDashboard.putData("NavX: ", this.navX);
```
Comment (dcowden) [inline Template2018/src/main/java/frc/team281/robot/subsystems/RealDriveSubsystem.java]: please use dataLogger.log() for this. that way, we can also send it other places if we'd like

File: Template2018/src/main/java/frc/team281/robot/subsystems/RealDriveSubsystem.java [hunk 2]
```diff
@@ -79,28 +95,36 @@ public void initialize() {
 
 		this.rightTalonSettings = TalonSettingsBuilder.inverted(leftTalonSettings, false, true);
 
+		if (this.navX != null) {
+			while (this.navX.isCalibrating()) {
+				try {
+					Thread.sleep(50);  // 50ms
```
Comment (dcowden) [inline Template2018/src/main/java/frc/team281/robot/subsystems/RealDriveSubsystem.java]: perhaps extract 50ms to a constant?

File: Template2018/src/main/java/frc/team281/robot/subsystems/RealDriveSubsystem.java (hunk 1 above)
Comment (mandrews281) [reply Template2018/src/main/java/frc/team281/robot/subsystems/RealDriveSubsystem.java]: You're right -- I'll fix it.  I guess old habits die hard :-).

File: Template2018/src/main/java/frc/team281/robot/subsystems/RealDriveSubsystem.java (hunk 2 above)
Comment (mandrews281) [reply Template2018/src/main/java/frc/team281/robot/subsystems/RealDriveSubsystem.java]: I don't think we are likely to use this constant anywhere else, so I'm inclined to leave it as is.

Comment (dcowden) [reply Template2018/src/main/java/frc/team281/robot/subsystems/RealDriveSubsystem.java]: i agree re-use is unlikely. however, Thread.sleep(NAVX_CALIBRATION_TIME_MS) is a more readable and cant become out of sync with the //50ms constant.

In the template i'm sensitive to setting good coding examples.  I'll merge now, we can make a cleanup pass through later.

Comment (mandrews281) [reply Template2018/src/main/java/frc/team281/robot/subsystems/RealDriveSubsystem.java]: There you go showing me how much more readable that it.  If you think that's going to work, you're right.  Another change coming.

Comment (dcowden) [reply Template2018/src/main/java/frc/team281/robot/subsystems/RealDriveSubsystem.java]: ;)

**Pull request merged**